Restructure of TLS Context (#46)
mulesoft/docs-anypoint-security: modules/ROOT/pages/configure-tls-context-rtf-lb.adoc

include::_attributes.adoc[]
endif::[]

All inbound traffic entering Anypoint Runtime Fabric is encrypted using TLS. When enabling inbound traffic, you must provide a valid TLS certificate. When you configure inbound traffic for Runtime Fabric, you can configure the TLS-server by importing a Privacy-Enhance Mail (PEM) or Java Keystore (JKS), or by importing a TLS context from secrets manager (advanced use case).

Before you can import the TLS Context from secrets manager, you need to add the TLS Context. 

== Add the TLS Context to a Secret Group

Adding a TLS context to your secret group involves supplying a name, target, security information, version, keystore and, optionally, truststore. 

. In the *Secret Groups* list view, select the secret group to which to add a TLS context. 
. Select *TLS Context* in the menu on the left, and click *Add TLS Context*. 
. In the Create TLS Context screen, add the required information:
+
* *Name* +
Enter a name for your TLS context. 
* *Target* +
Select *Anypoint Security* to validate the SSL handshake for Runtime Fabric. 
* *TLS Version* +
By default, TLS 1.2 and TLS 1.3 are selected. +
** TLS 1.1
** TLS 1.2
** TLS 1.3  
* *Keystore* +

> From the drop-down list, select the keystore to store the TLS context. +

Reviewer: You need to point back to a procedure for entering a keystore or provide in line for one example keystore type. In the spirit of replacing what we had before, I think you should add the example procedure for PEM in line so that we are not subtracting documentation by this restructuring.

Reviewer 2: Resolved?

> You can select keystores and truststores that are either PEM or non-PEM types. Supported types include: +

Reviewer: This should read "You can select keystore that are either ...."

*** JKS
*** PKCS #12
*** JCEKS 

> * *Truststore* +

Reviewer: Delete this section from here. All of this info belongs in the Client Auth secton. We are shooting for a procedure not functional doc.

Author: The client authentication topic is only a procedure that is specific to creating a TLS context. It doesn't have keystore and truststore info in it.

Optionally select a truststore to which to add the TLS context if you are using a truststore to store certificates trusted by the client. +
* *Expiration Date* +
Optionally select the expiration date for the certificate.
. Optionally xref:enable-client-authentication.adoc[enable client authentication]. 
. Optionally xref:cipher-suites.adoc[select ciphers]. 
. Click *Save*.

[[import_tls_rtf]]
== Import a TLS Context to Runtime Fabric

. In *Runtime Manager > Runtime Fabrics > Inbound Traffic*, select *Import from Secrets Manager*. 
. Select the environment the secret group that contains the TLS context is associated with.

> . Select the secret group where the TLS context is configured.

Reviewer 3: What does this mean? the customer needs to first select the environment and then select the secret group?

Author: Yes, and this is clear to the user.

= Configure a TLS Context for Anypoint Runtime Fabric
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

All inbound traffic entering Anypoint Runtime Fabric is encrypted using TLS. When enabling inbound traffic, you must provide a valid TLS certificate. When you configure inbound traffic for Runtime Fabric, you can configure the TLS server by importing a Privacy-Enhanced Mail (PEM) or Java Keystore (JKS) file, or by importing a TLS context from the secrets manager (advanced use case).

> Before you can import the TLS Context from the secrets manager, you must add the TLS context. 

Reviewer 2: ```suggestion
Before you can import the TLS context from the secrets manager, you must add the TLS context. 
```

== Add the TLS Context to a Secret Group

> Adding a TLS context to your secret group involves supplying a name, target, security information, version, keystore and, optionally, truststore. 

Reviewer 2: ```suggestion
Adding a TLS context to your secret group involves supplying a name, target, security information, version, keystore, and, optionally, truststore. 
```

> . In the *Secret Groups* list view, select the secret group to which you want to add a TLS context. 

Reviewer 2: Should "secret group" be "secrets group"? I initially changed it above, but now I am not sure.
To me, "secret group" sounds like a group that is secret, not a group that contains secrets.
I'll stop marking.

. Select *TLS Context* in the menu on the left, and click *Add TLS Context*. 
. In the Create TLS Context screen, add the required information:
+
* *Name* +
Enter a name for your TLS context. 
* *Target* +
Select *Anypoint Security* to validate the SSL handshake for Runtime Fabric. 
* *TLS Version* +
By default, TLS 1.2 and TLS 1.3 are selected. +
For information about the advantages of TLS 1.3, see xref:asm-secret-type-support-reference.adoc[Supported Secret Types].
* *Keystore* +
From the drop-down list, select the keystore to store the TLS context. +

> For information about adding a keystore to the secrets manager see xref:asm-secret-group-creation-task#add_keystore.adoc[Create a Secret Group]. +

Reviewer 2: ```suggestion
For information about adding a keystore to the secrets manager, see xref:asm-secret-group-creation-task#add_keystore.adoc[Create a Secret Group]. +
```

You can select keystores that are either PEM or non-PEM types. 
* Optionally select a truststore to which to add the TLS context if you are using a truststore to store certificates trusted by the client. +

> A trust store is required if you are going to enable client authentication.

Reviewer 2: ```suggestion
A truststore is required if you are going to enable client authentication.
```

* *Expiration Date* +
Optionally select the expiration date for the certificate.
. Optionally xref:enable-client-authentication.adoc[enable client authentication]. 
. Optionally xref:cipher-suites.adoc[select ciphers]. 
. Click *Save*.

[[import_tls_rtf]]
== Import a TLS Context to Runtime Fabric

. In *Runtime Manager > Runtime Fabrics > Inbound Traffic*, select one of the following options: +
* *Upload PEM* +

> For information about this option, see xref:runtime-fabric::enable-inbound-traffic.adoc[Enable Inbound Traffic on Anypoint Runtime Fabric]

Reviewer 2: ```suggestion
For information about this option, see xref:runtime-fabric::enable-inbound-traffic.adoc[Enable Inbound Traffic on Anypoint Runtime Fabric].
```

* *Upload JKS* +

> For information about this option, see xref:runtime-fabric::enable-inbound-traffic.adoc[Enable Inbound Traffic on Anypoint Runtime Fabric]

Reviewer 2: ```suggestion
For information about this option, see xref:runtime-fabric::enable-inbound-traffic.adoc[Enable Inbound Traffic on Anypoint Runtime Fabric].
```
